File: 382 - Perfection/382 - Perfection.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    printf("PERFECTION OUTPUT\n");
    int n = 0;
    cin >> n;
    while (n){
        int divs = 0;
        for (int i = 1; i < n;i++){if (n%i == 0){divs += i;}}
        if (divs == n){printf("%5d  PERFECT\n",n);}
        else{
            if(divs > n){printf("%5d  ABUNDANT\n",n);}
            else{printf("%5d  DEFICIENT\n",n);}
        }
        cin >> n;
    }
    printf("END OF OUTPUT\n");
    return 0;
}

File: 382 - Perfection/382 - Perfection.py
from sys import stdin as s
from math import sqrt

def main():
    print("PERFECTION OUTPUT")
    while True:
        flag = False
        numbers = list(map(int, s.readline().strip().split()))
        for number in numbers:
            if number == 0:
                flag = True
                break
            divs = 0
            root = int()
            res = "DEFICIENT"
            for i in range(1, number):
                if (number % i == 0): divs += i
            if (divs == number): res = "PERFECT"
            elif (divs > number): res = "ABUNDANT"
            print("{:>5}  ".format(number) + res)
        if flag:
            break
    print("END OF OUTPUT")
main()
